### ISSUE TITLE
no-unused-vars should have an importIgnorePattern

### ISSUE THREAD

**Issue by OoDeLally** [eslint/eslint#13347]
**The version of ESLint you are using.**
6.8.0

**The problem you want to solve.**

I have a custom `stringify` helper that I may add in any file for debugging purpose, and I'm tired of adding/removing the import constantly. I cannot leave the import in the file, as eslint would nag me constantly about the unused import.
I do not wish either to have `// eslint-disable-next-line no-unused-vars` in all my files.

Solution: Be able to disable `no-unused-vars` for _some_ unused imports.

Currently a similar feature exists for function arguments `argsIgnorePattern`. But it does not seem to apply for unused imports.

**Your take on the correct solution to problem.**

```
{ 
  no-unused-vars': ['warn', { 'importIgnorePattern': '^stringify$' }] 
}
```

**Are you willing to submit a pull request to implement this change?**
It depends on the complexity.

**Comment by kaicataldo** [comment]
Can you share a simplified code sample? I'm not currently understanding why you would want to create an import binding and then not use it.

**Comment by OoDeLally** [comment]
The rationale is exactly the same as ignoring an unused function argument.


Let's say I have the following file
```js
const myFunction = () => {
	const value = getSomeValue();
	return value;
};
```js
Now I notice I suspect an issue with `getSomeValue`, so I add a log after it:
```ts
import { stringify } from './helpers';

const myFunction = () => {
	const value = getSomeValue();
	console.log(stringify(value));
	return value;
};
```

Great, I can see what `getSomeValue` returns, and I've found the mistake. I can now remove the log:
```js
import { stringify } from './helpers';

const myFunction = () => {
	const value = getSomeValue();
	return value;
};
```

Of course, now `eslint` nags me about `stringify` not being used.
Sure, I could remove it, but being in the process of building my thing, I know I will have to reuse it in the very near future.

Now the current solution is to use:

```js
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { stringify } from './helpers';
```

What could be even better is being able to simply ignore that `stringify` is not used.

**Comment by mdjermanovic** [comment]
> The rationale is exactly the same as ignoring an unused function argument.

This doesn't look like the same thing to me. Would you want to commit a change with an unused import?

**Comment by OoDeLally** [comment]
> This doesn't look like the same thing to me. Would you want to commit a change with an unused import?

Yes, I would like to commit it with an unused import. 
Not every commit, and further, not every repo is public and must be canonically clean.
It really boils down to the convenience / cleanliness ratio, and there are some situation where I would prefer it as such.

**Comment by anikethsaha** [comment]
It looks like it can be good for a custom [plugin](https://eslint.org/docs/developer-guide/working-with-plugins) / [rule](https://eslint.org/docs/developer-guide/working-with-rules).

**Comment by kaicataldo** [comment]
I don't really understand why this is specific to imports, as the rule will do the same thing for unused variable declarations as well. 

My personal take: I just ignore these kinds of errors when I'm in the process of writing code. If you're running ESLint as part of your bundler and this is preventing it from compiling, I'd recommend not running ESLint that way for this exact reason.

**Comment by francisminu** [comment]
I am facing an issue along the same lines.
I have a .ts file with the import:
    import { PIDateRangeDataModel } from './pi-date-range-data.model';

and this is just a class file and I have a variable of type PIDateRangeDataModel
    countDate: PIDateRangeDataModel;

I keep getting the Eslint error:
     2:10  error  'PIDateRangeDataModel' is defined but never used  no-unused-vars


Could someone please let me know how this can be resolved without setting ignore for the rule on all the files?

**Comment by mdjermanovic** [comment]
@francisminu 

For typescript code you should use [`@typescript-eslint/no-unused-vars`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars.md) or [`@typescript-eslint/no-unused-vars-experimental`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars-experimental.md) instead of the core `no-unused-vars` rule.
